contracts/DataFeeds/TimeMedianDataFeed.sol

```diff
@@ -0,0 +1,145 @@
+pragma solidity ^0.4.24;
+
+import "tidbit/contracts/DataFeedOracles/DataFeedOracleBase.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract TimeMedianDataFeed is DataFeedOracleBase {
+  using SafeMath for uint;
+
+  mapping(uint => bool) dateAlreadyAccountedFor; // transitory data structure useful only during function call medianizeByDates
+
+  function medianizeByTimeframe(uint startDate, uint endDate)
+    public
+    view
+    returns (bytes32 medianValue)
+  {
+    require(startDate < endDate, 'startDate must be less than endDate');
+    uint endIndex;
+    uint startIndex;
+
+    // find end date within dates array
+    for (uint i = dates.length - 1; i > 0 && endIndex == 0; i--) {
+      if (dates[i] <= endDate) {
+        endIndex = i; // ONLY dates before the end date
+      }
+    }
+
+    // require endIndex is more than 0
+    require(endIndex > 0, 'dates outside of range of datafeed results');
+    require(dates[endIndex] >= startDate, 'dates outside of range of datafeed results');
+
+    // find start date within dates array
+    for (uint j = endIndex; j >= 0 && startIndex == 0; j--) {
+      if (dates[j] < startDate) {
+        startIndex = j + 1; // no dates before the startDate
+      }
+    }
+
+    medianValue = medianizeByIndices(startIndex, endIndex);
+  }
+
+  function medianizeByIndices(uint startIndex, uint endIndex)
+    public
+    view
+    returns (bytes32 medianValue)
+  {
+    require(startIndex <= endIndex, 'startIndex must be less than or equal to endIndex');
+    require(startIndex > 0, 'startIndex must be greater than 0');
+
+    uint[] memory selectedDates = _sliceDatesArray(startIndex, endIndex);
+    uint[] memory partitionedDates = _partitionDatesByResult(selectedDates);
+    medianValue = _medianizeByDates(partitionedDates);
+  }
+
+  function medianizeByDates(uint[] orderedDates)
+    public
+    view
+    returns (bytes32 medianValue)
+  {
+    for (uint i = 0; i < orderedDates.length; i++) {
```

Comment:
since this is a view function, I don't think the duplicate check is needed

Reply:
on line 63? So it's okay for people to put in duplicate dates on this>

Reply:
it would be good to have a duplicate check, but it seems odd that it's altering state in a `view` function

Reply:
it might make sense to remove all restrictions from this (except maybe `isResultSetFor`). It's already possible to omit dates from the ordered array. This could be a public view function that takes any set of dates and just returns the middle value.